packages/service-library/src/servicelib/file_utils.py

```diff
@@ -60,10 +62,59 @@ async def create_sha256_checksum(
 
 async def _eval_hash_async(
     async_stream: AsyncStream,
-    hasher: "hashlib._Hash",  # noqa: SLF001
+    hasher: "hashlib._Hash",
     chunk_size: ByteSize,
 ) -> str:
     while chunk := await async_stream.read(chunk_size):
         hasher.update(chunk)
     digest = hasher.hexdigest()
     return f"{digest}"
+
+
+def _get_file_properties(path: Path) -> tuple[float, int]:
+    stats = path.stat()
+    return stats.st_mtime, stats.st_size
+
+
+def _get_directory_snapshot(path: Path) -> dict[str, tuple[float, int]]:
+    return {
+        f"{x.relative_to(path)}": _get_file_properties(x)
```

Comment:
MINOR: I understand your point about using `_` as the interation variable, but could you please name the variable with something related to what you are iterating? E.g. can you call here `p` instead of `x`?

Actually, the fact that you use a random name, like `x`, indicates that the statement readability 
That is the reason why some people use `_`

```python
one_way    = { f"{_.relative_to(path)}": get_properties(_) for _ in path.rglob("*") if _.is_file()  }

equivalent = { f"{p.relative_to(path)}": get_properties(p) for p in path.rglob("*") if p.is_file() }

```